modules/appeals_api/app/workers/appeals_api/pdf_submit_job.rb

```diff
@@ -89,20 +96,26 @@ def handle_upload_error(appeal, e)
       log_upload_error(appeal, e)
       appeal.update_status(status: 'error', code: e.code, detail: e.detail)
 
-      if e.code == 'DOC201' || e.code == 'DOC202'
-        notify(
-          {
-            'class' => self.class.name,
-            'args' => [appeal.id, appeal.class.to_s, appeal.created_at.iso8601],
-            'error_class' => e.code,
-            'error_message' => e.detail,
-            'failed_at' => Time.zone.now
-          }
-        )
-      else
-        # allow sidekiq to retry immediately
-        raise
-      end
+      # Get the underlying error from EMMS
+      emms_resp_status = e.detail[/(?<=Upstream|Downstream status: )\d*/]
+
+      # if we can't get the EMMS call's response status,
+      # reraise the exception so sidekiq will retry
+      raise if emms_resp_status.blank?
```

Comment:
This line seems too risky to me. We had a recent error when submitting, `Downstream status: Errors: ERR-EMMS-FAILED, Log Date cannot be in the future...` When we reached out to the CMP folks, they confirmed that the PDF had already been successfully submitted. The regex match on L100 returns a blank string for that error message, so in that case, the job would have retried and resubmitted the PDF, which we wouldn't want. I think we need to err on the side on caution and only retry if the error is a confirmed 401, 429, 500, or 503 error.

Reply:
Looking through more example error `details`, I think partial string matching for the response code in the error `detail` itself is more fraught than I originally thought. Could we add a new accessible attribute to `CentralMail::UploadError` that holds the response code returned from CMP and is read in `PdfSubmitJob` to determine whether to retry or not?

Reply:
lib/central_mail/utilities.rb#map_errors is passed the raw http status returned by EMMS and puts it in the detail, the only way I can think of that the detail string in the exception that is being parsed on line 100 would fail is if the raw call to EMMS did not return a status....

whether we track the raw http status returned from EMMS in the exception detail string like we are doing now OR we add a new attribute to CentralMail::UploadError that holds the WMMS status..in either case we would not have a status code to populate the detail staring nor a specific exception attribute...

Reply:
Just closing the loop on this conversation. We chatted about the solution in person and came to the conclusion to only retry if the status code was in the list of retry-able codes, and to add the status code as an attribute on the `UploadError` (both of which were implemented).